ansible_base/lib/routers/association_resource_router.py

```diff
@@ -0,0 +1,187 @@
+import logging
+
+from django.utils.translation import gettext as _
+from rest_framework import routers, serializers, status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+logger = logging.getLogger('ansible_base.lib.routers.association_resource_router')
+
+
+class QuerySetMixinBase:
+    def get_queryset(self):
+        parent_pk = self.kwargs['pk']
+        parent_model = self.backward_related_qs.model
+
+        try:
+            parent_instance = parent_model.objects.get(pk=parent_pk)
+        except parent_model.DoesNotExist:
+            return parent_model.objects.none()
+
+        if issubclass(self.__class__, AssociateMixin):
+            child_queryset = getattr(parent_instance, self.association_fk).all()
```

Comment:
nit: Can we let the subclass itself determine whether to use `self.association_fk` or `self.related_fk`? Like have it be a class attribute or something, rather than introspecting the subclass of this base class.

Reply:
Fixed... ish. Has a bug that I need to track down but it now it uses class property.